src/main/java/io/github/hapjava/characteristics/impl/base/IntegerCharacteristic.java

```diff
@@ -62,17 +62,21 @@ public IntegerCharacteristic(
     return super.makeBuilder(iid)
         .thenApply(
             builder -> {
-              return builder
-                  .add("minValue", minValue)
-                  .add("maxValue", maxValue)
-                  .add("minStep", 1)
-                  .add("unit", unit);
+              builder.add("minValue", minValue).add("maxValue", maxValue).add("minStep", 1);
+              if (this.unit != null) {
+                builder.add("unit", unit);
+              }
+              return builder;
             });
   }
 
   @Override
   protected CompletableFuture<Integer> getValue() {
-    return getter.map(integerGetter -> integerGetter.get()).get();
+    if (getter.isPresent()) {
```

Comment:
good point with null, 
replacing get with orElse should do the trick
`return getter.map(integerGetter -> integerGetter.get()).orElse(null);`

Reply:
Thank you, I was struggling with that one